src/components/flame-graph/Canvas.js

```diff
@@ -0,0 +1,239 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import * as React from 'react';
+import seedrandom from 'seedrandom';
+import withChartViewport from '../shared/chart/Viewport';
+import ChartCanvas from '../shared/chart/Canvas';
+import TextMeasurement from '../../utils/text-measurement';
+
+import type { Thread } from '../../types/profile';
+import type { CssPixels, UnitIntervalOfProfileRange } from '../../types/units';
+import type {
+  FlameGraphTiming,
+  FlameGraphDepth,
+  IndexIntoFlameGraphTiming,
+} from '../../profile-logic/flame-graph';
+import type { Action, ProfileSelection } from '../../types/actions';
+import type { CallNodeInfo } from '../../types/profile-derived';
+
+type Props = {
```

Comment:
I reworked the way we handle the Props for React components, so #709 will break this when it lands.